lib/pos_epoch_complete_hook.go

```diff
@@ -266,13 +273,15 @@ func (bav *UtxoView) runEpochCompleteEpochRollover(epochNumber uint64, blockHeig
 	return nil
 }
 
-func (bav *UtxoView) generateAndSnapshotValidatorSet(epochNumber uint64) ([]*ValidatorEntry, error) {
+func (bav *UtxoView) generateAndSnapshotValidatorSet(epochNumber uint64, maxValidatorSet []*ValidatorEntry,
+) []*ValidatorEntry {
```

Comment:
You can assume that the validator set passed in here is the netire validator set. This helper function is coupled and intended to be used from one place anyway. There's no need to return the validator set from this function because it's the same validator entries that were passed in
```suggestion
func (bav *UtxoView) snapshotValidatorSet(epochNumber uint64, validatorSet []*ValidatorEntry,
) {
```